packages/ag-charts-community/src/chart/themes/chartTheme.ts

```diff
@@ -396,28 +396,30 @@ export class ChartTheme {
         );
     }
 
-    templateTheme<T>(themeTemplate: T): T {
-        const themeInstance = deepClone(themeTemplate);
-        const params = this.getTemplateParameters();
-
-        jsonWalk(themeInstance, (node: any) => {
-            if (isArray(node)) {
-                for (let i = 0; i < node.length; i++) {
-                    const symbol = node[i];
-                    if (params.has(symbol)) {
-                        node[i] = params.get(symbol);
-                    }
+    private templateThemeFn(node: any, _other: any, params?: Map<any, any>) {
+        if (isArray(node)) {
+            for (let i = 0; i < node.length; i++) {
+                const symbol = node[i];
+                if (typeof symbol === 'symbol' && params?.has(symbol)) {
+                    node[i] = params.get(symbol);
                 }
-            } else {
-                for (const [name, value] of Object.entries(node)) {
-                    if (params.has(value)) {
-                        node[name] = params.get(value);
-                    }
+            }
+        } else {
+            for (const [name, value] of Object.entries(node)) {
+                if (typeof value === 'symbol' && params?.has(value)) {
+                    node[name] = params.get(value);
                 }
             }
-        });
+        }
+    }
+
+    templateTheme<T>(themeTemplate: T, clone = true): T {
+        const themeInstance = clone ? deepClone(themeTemplate) : themeTemplate;
+        const params = this.getTemplateParameters();
+
+        jsonWalk(themeInstance, this.templateThemeFn, undefined, undefined, params);
```

Comment:
```suggestion
        jsonWalk(themeInstance, this.templateThemeFn, undefined, params);
```
Redundant `undefined` arguments?

Reply:
Good spot - I'm intrigued that this didn't flag up on any tests 🤔

Reply:
Actually this is correct, we want to pass the same `ctx` object into the handling function

Reply:
Reverted

Reply:
Oh, so `params` is the `ctx`...